tests/issue/issue.cpp

```diff
@@ -604,3 +615,154 @@ TEST(Issue, undefinedIssueUrl)
     EXPECT_EQ(e, issue->url());
     EXPECT_EQ(e, issue->referenceHeading());
 }
+
+#endif
+
+TEST(Issue, setGetByObjectOrItem)
```

Comment:
@agarny Here's what I'm struggling with! Inside the `Issue::Impl` is two fields:
- `mAnyItem` which is an `std::any`
- `mItemType` which is one of the `ItemType` enums.
At the moment I'm not doing any switching, it's all written as if I will only ever store `Component` types inside the any class (for my debugging sanity).  So.  

I can add and retrieve an item (here it's a `Component`) into the `mPimpl->mAnyItem` item, which is a `std::any` class.  When you add anything, they're all stored in the `mAnyItem` slot.
When you retrieve anything specifically (eg: `issue->component()`) it will do the casting first and return a component pointer.  But you can also choose to retrieve the std::any item itself, so that you can do the casting later based on the value of the `mItemType`.  It's this external casting which is failing.  I've checked that the same pointers and values are being stored, but when I call `std::any_cast` in the test (ie: outside the `Issue`) class, it fails.  The operation is identical to what's happening internal to the `Issue` class in order to return the component directly.

Reply:
Just a look and... it seems to be working fine for me on macOS. I mean that line 644 down there is fine... So... what am I missing here?

Reply:
:O Really?? I'm not sure if that's good news or bad news ... Let me look again ...